docs/serverless/functions.md

```diff
@@ -0,0 +1,175 @@
+
+### [.NET CLI](#tab/dotnet-cli)
+
+```dotnetcli
+dotnet add package Aspire.Hosting.Azure.Functions --prerelease
+```
+
+### [PackageReference](#tab/package-reference)
+
+```xml
+<PackageReference Include="Aspire.Hosting.Azure.Functions"
+                  Version="*" />
+```
+
+---
+
+For more information, see [dotnet add package](/dotnet/core/tools/dotnet-add-package) or [Manage package dependencies in .NET applications](/dotnet/core/tools/dependencies).
+
+### Add Azure Functions resource
+
+In your app host project, call `AddAzureFunctionsProject` on the `builder` instance to add an Azure Functions resource:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var functions = builder.AddAzureFunctionsProject<Projects.ExampleFunctions>("functions")
+                       .WithExternalHttpEndpoints();
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(functions)
+       .WaitFor(functions);
+
+// After adding all resources, run the app...
+```
+
+When .NET Aspire adds an Azure Functions project resource the app host, as shown in the preceding example, the `functions` resource can be referenced by other project resources. The <xref:Aspire.Hosting.ResourceBuilderExtensions.WithReference%2A> method configures a connection in the `ExampleProject` named `"functions"`. If the Azure Resource was deployed and it exposed an HTTP trigger, its endpoint would be external due to the call to <xref:Aspire.Hosting.ResourceBuilderExtensions.WithExternalHttpEndpoints*>. For more information, see [Reference resources](../fundamentals/app-host-overview.md#reference-resources).
+
+### Add Azure Functions resource with host storage
+
+If you want to modify the default host storage account that the Azure Functions host uses, call the `WithHostStorage` method on the Azure Functions project resource:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var storage = builder.AddAzureStorage("storage")
+                     .RunAsEmulator();
+
+var functions = builder.AddAzureFunctionsProject<Projects.ExampleFunctions>("functions")
+                       .WithHostStorage(storage);
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(functions)
+       .WaitFor(functions);
+
+// After adding all resources, run the app...
+```
+
+The preceding code relies on the [📦 Aspire.Hosting.Azure.Storage](https://www.nuget.org/packages/Aspire.Hosting.Azure.Storage) NuGet package to add an Azure Storage resource that runs as an emulator. The `storage` resource is then passed to the `WithHostStorage` API, explicitly setting the host storage to the emulated resource.
+
+> [!NOTE]
+> The `StorageAccountContributor` role is required for deployed instances, it's up to the user to manually configure this role with their resource when they're publishing. This role assignment is included automatically for the implicitly generated host storage.
```

Comment:
This should clarify that you only need to manually configure if you are _not_ using the implicit host storage.